packages/dialect-react-sdk/context/DialectContext/Sdk/index.ts

```diff
@@ -22,7 +24,33 @@ function useDialectSdk(
     encryptionKeysStore,
     backends = EMPTY_ARR,
   } = config;
-  const { adapter, connected } = DialectWallet.useContainer();
+  const { adapter, connected, initiateConnection } =
+    DialectWallet.useContainer();
+
+  // checks if wallet is already authorized to skip not authorized screen
+  useEffect(
+    function preValidateSdkToken() {
+      if (!dialectCloud) return;
+      if (!dialectCloud.tokenStore) return;
+      if (!adapter.publicKey) return;
+      // extract token store
+      let tokenStore: TokenStore;
+      if (typeof dialectCloud.tokenStore === 'string') {
+        if (dialectCloud.tokenStore !== 'local-storage') return;
+        tokenStore = TokenStore.createLocalStorage();
+      } else {
+        tokenStore = dialectCloud.tokenStore;
+      }
+
+      const token = tokenStore.get(adapter.publicKey);
+      if (!token) return;
+      const tokenValid = Auth.tokens.isValid(token);
+      if (!tokenValid) return;
+      initiateConnection();
+    },
+    [config, adapter, initiateConnection]
```

Comment:
Effect is missing `dialectCloud` in `deps` array.

Reply:
yea, but it's not needed, since it has `config`, which contains dialectCloud. does it make sense?

Reply:
yes, but ESLint would throw a warning here. I understand your point, but would replacing `config` with `dialectCloud` break this effect? If so, why?

Reply:
technically shouldn't break anything, replaced to `dialectCloud`